osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -120,7 +120,7 @@ public override void OnEntering(IScreen last)
             var track = Beatmap.Value.Track;
             var metadata = Beatmap.Value.Metadata;
 
-            if (last is Intro && track != null)
+            if (last is Intro && track != null && !Game.MusicController.UserRequestedPause)
```

Comment:
Resolve the `MusicController` as a private property instead.

```
[Resolved]
private MusicController musicController { get; set; }
```

Reply:
We also need to access the musicController in the `SongSelect` class. Should I also put these two lines there or resolving it in the `OsuScreen` (the base class for both `MainMenu` and `SongSelect`) as a protected property to avoid code repetition would be better?

Also, sorry for messing up the `Editor` with `SongSelect` in my previous comment. I surely caused a lot of confusion...

Reply:
resolve via DI in both places.

Reply:
Got it!